x-pack/plugin/otel-data/src/main/resources/index-templates/README.md

```diff
@@ -0,0 +1,7 @@
+Mappings and settings that must not be overridden are to be defined
+in the index templates, as these will take precedence over component
+template mappings and settings. This includes:
+
+ - `index.default_pipeline`
+ - `index.final_pipeline`
+ - values for `data_stream.*`
```

Comment:
> Is that an intentional departure from apm-data?

Not really. I think this mostly comes from https://github.com/elastic/elasticsearch/pull/104455 - and honestly I was not aware of this difference. 

The readme file in this form is definitely wrong - we don't set  pipeline related things, and as you point out, `data_stream.*` are set in component templates. So I removed `README.md`

What should we do? Should we move the `data_stream.*` into index templates?

The only thing we'd set is probably `data_stream.type` value. But that's handled in `logs-otel@mappings`, `metrics-otel@mappings`, and in `traces@mappings` (that's in the core). The default value for `data_stream.type` can be already defined in those component templates, so I feel it's a better fit to already define it there, and with that, there isn't really anything to set in the index templates.

So if I understand correctly, this is more like, there is nothing left to set there at this point.

Reply:
> The only thing we'd set is probably data_stream.type value.

The reason for setting constant values in the index template is to prevent them from being accidentally overridden by `@custom` component templates. The index template takes precedence over component templates, so anything that must not be overridden (e.g. `data_stream.type`) should ideally be defined in the index template.

Reply:
BTW I wonder if we should fold all of the `<type>-otel@mappings` component templates into their related index templates. There's no reuse going on, so it would be simpler that way. Maybe in a followup?

Reply:
> The reason for setting constant values in the index template is to prevent them from being accidentally overridden by @custom component templates. The index template takes precedence over component templates, so anything that must not be overridden (e.g. data_stream.type) should ideally be defined in the index template.

Thanks for pointing this out - that is a really good argument for setting it in the index template. I pushed it in
https://github.com/elastic/elasticsearch/pull/111091/commits/e1adb6ac8acc3481ecc27ce120cb553fa13554a4 and also added some tests to cover this.

> BTW I wonder if we should fold all of the <type>-otel@mappings component templates into their related index templates. There's no reuse going on, so it would be simpler that way. Maybe in a followup?

Maybe the `-otel@custom` component templates would be a challenge as Felix pointed out in https://github.com/elastic/elasticsearch/pull/111091#discussion_r1713397252 - but we can definitely iterate on this, I'd prefer to do so in a follow up.

Reply:
Sounds good to me, thank you!